File: ios/Developer Wallet/UI.swift
//
//  UI.swift
//  Developer Wallet
//
//  Created by Daniel Leping on 24/01/2023.
//

import Foundation
import TesseractTransportsService
import CWallet

public class UI {
    func requestUserConfirmation(request: Request) async -> Result<Bool, WalletError> {
        fatalError("No UI is implemented in the Wallet. Use extension.")
    }
    
    func toCore() -> SUI {
        SUI(ui: self)
    }
}

File: common/src/ios/ui/ui.rs
use async_trait::async_trait;

use crate::request::Request;
use crate::ui::UIProtocol;
use crate::error::Result as DWResult;
use super::sui::SUI;

pub struct UI {
    swift: SUI
}

impl UI {
    pub (crate) fn new(sui: SUI) -> Self {
        UI {
            swift: sui
        }
    }
}

#[async_trait]
impl UIProtocol for UI {
    async fn request_user_confirmation<R: Request>(&self, request: R) -> DWResult<bool> {
        Ok(true)
    }
}